bundles/core/org.openhab.core/src/main/java/org/openhab/core/types/UnDefType.java

```diff
@@ -34,4 +34,20 @@ public String format(String pattern) {
 		return String.format(pattern, this.toString());
 	}
 
+	/**
+	 * Returns the enum constant with the specified name.
+	 */
+	public static UnDefType parse(String name) {
+		if (name == null) {
+			throw new NullPointerException("Name is null");
+		}
+		if (UNDEF.toString().equals(name)) {
+			return UNDEF;
+		} else if (NULL.toString().equals(name)) {
+			return NULL;
+		}
+		throw new IllegalArgumentException("No enum constant "
+				+ UnDefType.class.getCanonicalName() + "." + name);
+	}

```

Comment:
That's great, but the specific problem the change is intended to fix is still not clear to me. I have many JSR223/Jython scripts and they work as expected even without this change. I agree with @watou that this should be broken out into a separate PR.